File: c_cpp_migration/cpp_project/gameplay/quest/QUEST_CATALOG.cpp
#ifndef QUEST_CATALOG_HPP
#define QUEST_CATALOG_HPP

#include <iostream>
#include <string>
#include <vector>
#include <memory>
#include <unordered_map>
#include <cmath>
#include <cstdlib>
#include <algorithm>
#include <optional>
#include <functional>
#include <random>

namespace lostjump {

// Function declaration
auto QUEST_CATALOG();

} // namespace lostjump

#endif // QUEST_CATALOG_HPP

// Implementation
namespace lostjump {

#include <iostream>
#include <string>
#include <vector>
#include <memory>
#include <unordered_map>
#include <cmath>
#include <cstdlib>
#include <algorithm>
#include <optional>
#include <functional>
#include <random>
#include "contracts.js.hpp"
#include "mainQuests.js.hpp"
#include "npc_sidequests.js.hpp"






export const QUEST_CATALOG = [
  ...ACT1_MAIN_QUESTS,
  ...WORLD_CONTRACTS,
  ...WORLD_NPC_SIDEQUESTS,
];

export const QUESTS_BY_ID = Object.fromEntries(QUEST_CATALOG.map([](auto& item){ return (q; }) => [q.id, q]));


} // namespace lostjump

File: c_cpp_migration/legacy_js/src/gameplay/quest/questCatalog.js
// gameplay/quest/questCatalog.js
import { ACT1_MAIN_QUESTS } from "../../data/content/acts/act1/mainQuests.js";
import { WORLD_CONTRACTS } from "../../data/content/world/quests/contracts.js";
import { WORLD_NPC_SIDEQUESTS } from "../../data/content/world/quests/npc_sidequests.js";

export const QUEST_CATALOG = [
  ...ACT1_MAIN_QUESTS,
  ...WORLD_CONTRACTS,
  ...WORLD_NPC_SIDEQUESTS,
];

export const QUESTS_BY_ID = Object.fromEntries(QUEST_CATALOG.map((q) => [q.id, q]));
